mlflow/utils/environment.py

```diff
@@ -442,11 +444,31 @@ def _is_mlflow_requirement(requirement_string):
             # Try again with the per-requirement options removed
             return Requirement(requirement_specifier).name.lower() == "mlflow"
         except InvalidRequirement:
-            return False
 
-        # TODO: Return True if `requirement_string` represents a project directory for MLflow
-        # (e.g. '/path/to/mlflow') or git repository URL (e.g. 'https://github.com/mlflow/mlflow').
-        return False
+            # Support defining branch dependencies for local builds or direct GitHub builds
+            # from source.
+            # Example: mlflow @ git+https://github.com/mlflow/mlflow@branch_2.0
+            repository_matches = ["/mlflow", "mlflow@git"]
+
+            return any(
+                match in requirement_string.replace(" ", "").lower() for match in repository_matches
+            )
+
+
+def _generate_mlflow_version_pinning():
+    """
+    Determines the current MLflow version that is installed and adds a pinned boundary version range
+    for mlflow. The upper bound is a cap on the next major revision. The lower bound is a cap on
+    the current installed minor version(i.e., 'mlflow<3,>=2.1')
+    :return: string for MLflow dependency version
+    """
+    mlflow_environment_version = Version(VERSION)
+    current_major_version = mlflow_environment_version.major
+    current_minor_version = mlflow_environment_version.minor
```

Comment:
```suggestion
    mlflow_version = Version(VERSION)
    current_major_version = mlflow_version.major
    current_minor_version = mlflow_version.minor
```

Nit: can we remove `_environment`? `mlflow_environment_version` sounds like the version of an environment where mlflow is installed.

Reply:
of course! Pushing the change in a few minutes :)